### PR TITLE
[wasm] HTTP tests on  NodeJS

### PR DESCRIPTION
**Pull request by maraf**
- Use `node-fetch` and `node-abort-controller` polyfills.
- Always use `node_fs` on node in `fetch_like` to get runtime assets.
- Disable tests that are not supported on node.
  - Multiple HTTP header value concatenation.
  - HTTP header ordering.
  - [HTTP response streaming](https://github.com/dotnet/runtime/issues/65429).

### CODE REVIEW

**Review by lewing — COMMENTED** [review]
I think we should consider a platform check for node here, I don't like disabling the tests on browser when the work for the actual browser case.

File: src/mono/wasm/runtime/polyfills.ts [hunk 1]
```diff
@@ -5,10 +5,7 @@ let node_url: any | undefined = undefined;
 
 export async function fetch_like(url: string): Promise<Response> {
     try {
-        if (typeof (globalThis.fetch) === "function") {
-            return globalThis.fetch(url, { credentials: "same-origin" });
-        }
-        else if (ENVIRONMENT_IS_NODE) {
+        if (ENVIRONMENT_IS_NODE) {
```
Comment (maraf) [inline src/mono/wasm/runtime/polyfills.ts:8]: @pavelsavara Are you ok with this change? When I added the fetch polyfill, it tried to use it to download runtime assets.

File: src/libraries/System.Net.Http/tests/FunctionalTests/package.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "system.net.websockets.client.tests",
+  "private": true,
+  "dependencies": {
+    "node-abort-controller": "^3.0.1",
```
Comment (pavelsavara) [inline src/libraries/System.Net.Http/tests/FunctionalTests/package.json:5]: We need exact versions. Also please check their dependencies, if they are safe for CI.

File: src/mono/wasm/test-main.js [hunk 3]
```diff
@@ -360,15 +361,25 @@ if (is_node) {
...
+                globalThis.fetch = moduleExport.default;
+                globalThis.Headers = moduleExport.Headers;
+                globalThis.Request = moduleExport.Request;
+                globalThis.Response = moduleExport.Response;
+            } else if(moduleName == "node-abort-controller") {
+                message += ' and attaching to global';
+                globalThis.AbortController = moduleExport.AbortController;
             }
-
+
```
Comment (pavelsavara) [inline src/mono/wasm/test-main.js:382]: whitespace

File: src/libraries/System.Net.Http/tests/FunctionalTests/package.json (hunk 2 above)
Comment (maraf) [reply src/libraries/System.Net.Http/tests/FunctionalTests/package.json:5]: There are 3 dependencies 
- [whatwg-url](https://www.npmjs.com/package/whatwg-url)
- [webidl-conversions](https://www.npmjs.com/package/webidl-conversions)
- [tr46](https://www.npmjs.com/package/tr46)

Audit found 0 vulnerabilities.

**Review by pavelsavara — APPROVED** [review]

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs [hunk 4]
```diff
@@ -92,6 +92,8 @@ public static partial class PlatformDetection
         public static bool IsBrowserDomSupportedOrNotBrowser => IsNotBrowser || IsBrowserDomSupported;
         public static bool IsNotBrowserDomSupported => !IsBrowserDomSupported;
         public static bool IsWebSocketSupported => IsEnvironmentVariableTrue("IsWebSocketSupported");
+        public static bool IsNodeJS => IsEnvironmentVariableTrue("IsNodeJS");
```
Comment (radical) [inline src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:95]: Just curious - is it possible to detect this without using environment variables?

File: src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http1.cs [hunk 5]
```diff
@@ -16,7 +16,7 @@ public class HttpClientHandlerTest_Http1 : HttpClientHandlerTestBase
     {
         public HttpClientHandlerTest_Http1(ITestOutputHelper output) : base(output) { }
 
-        [Fact]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotNodeJS))]
```
Comment (radical) [inline src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http1.cs:19]: Why is this disabled?

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs (hunk 4 above)
Comment (maraf) [reply src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:95]: We compute that value in the [`test-main.js`](https://github.com/dotnet/runtime/blob/158f0897958f857d3fe2b142371a3831a7d4f023/src/mono/wasm/test-main.js#L11). Once we have an interop API that is referenceable from `TestUtilities`, we could invoke the same expression here.

File: src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http1.cs (hunk 5 above)
Comment (maraf) [reply src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http1.cs:19]: Because `node-fetch` polyfill doesn't sort headers.

**Pull request merged**